cadence/contracts/contract-templates/AxelarFungibleToken.cdc

```diff
@@ -0,0 +1,271 @@
+                        })
+                    )
+                case Type<FungibleTokenMetadataViews.TotalSupply>():
+                    return FungibleTokenMetadataViews.TotalSupply(totalSupply: AxelarFungibleToken.totalSupply)
+            }
+            return nil
+        }
+    }
+
+    /// Function that creates a new Vault with a balance of zero
+    /// and returns it to the calling context. A user must call this function
+    /// and store the returned Vault in their storage in order to allow their
+    /// account to be able to receive deposits of this token type.
+    ///
+    /// @return The new Vault resource
+    ///
+    pub fun createEmptyVault(): @Vault {
+        return <-create Vault(balance: 0.0)
+    }
+
+    pub resource Administrator {
+
+        /// Function that creates and returns a new minter resource
+        ///
+        /// @param allowedAmount: The maximum quantity of tokens that the minter could create
+        /// @return The Minter resource that would allow to mint tokens
+        ///
+        pub fun createNewMinter(): @Minter {
+            emit MinterCreated()
+            return <-create Minter()
+        }
+
+        /// Function that creates and returns a new burner resource
+        ///
+        /// @return The Burner resource
+        ///
+        pub fun createNewBurner(): @Burner {
+            emit BurnerCreated()
+            return <-create Burner()
+        }
+    }
+
+    access(account) fun getAdminCapability(): @Administrator{
+        return <- create Administrator()
+    }
+
+
+    /// Resource object that token admin accounts can hold to mint new tokens.
+    ///
+    pub resource Minter {
+
+        pub fun mintTokens(amount: UFix64): @AxelarFungibleToken.Vault {
+            pre {
+                amount > 0.0: "Amount minted must be greater than zero"
+            }
+            AxelarFungibleToken.totalSupply = AxelarFungibleToken.totalSupply + amount
+            emit TokensMinted(amount: amount)
+            return <-create Vault(balance: amount)
+        }
+
+        init() {
+        }
+    }
+
+    /// Resource object that token admin accounts can hold to burn tokens.
+    ///
+    pub resource Burner {
+
+        /// Function that destroys a Vault instance, effectively burning the tokens.
+        ///
+        /// Note: the burned tokens are automatically subtracted from the
+        /// total supply in the Vault destructor.
+        ///
+        /// @param from: The Vault resource containing the tokens to burn
+        ///
+        pub fun burnTokens(from: @FungibleToken.Vault) {
+            let vault <- from as! @AxelarFungibleToken.Vault
+            let amount = vault.balance
+            destroy vault
+            emit TokensBurned(amount: amount)
+        }
+    }
+
+    init(tokenName: String, symbol: String) {
+        self.name = tokenName
+        self.symbol = symbol
+        self.totalSupply = 1000.0
+        self.VaultStoragePath = /storage/exampleTokenVault
```

Comment:
Good catch, Fixed!  Oversight on my part!

Reply:
Quick question.
I had to switch the tokenName to token symbol instead since the token name could possibly include spaces.  Is there a way to do .replace in cadence to change all of the spaces to underscores?  Or is it ok to keep it with the just the symbol?  I figure there wont be many cases with tokens with the same symbol under the same address.